### PR TITLE
Document impact of export annotations on dead code elimination

### PR DESCRIPTION
**Pull request by OlivierBlanvillain**
Hi @sjrd :). I'm not sure everything I wrote is accurate, but I feel that there is some documentation missing on this point.

### CODE REVIEW

**Comment by sjrd** [comment]
> Export annotations are the entry point for dead code elimination.

This is true. But I'm not sure what the point of the rest of the paragraph is. Although technically correct, it is mostly misleading. Your library is going to require the Scala standard lib anyway, including `Option` or `List`, unless you're going to *great* lengths not to use anything from the Scala lib in your entire program (i.e., that never happens).

> Library authors are also encouraged to limit their use export annotations to functions returning subtypes of `js.Any`, as doing otherwise will impact the size of binaries generated for all the library users.

That's not true. Export what you need. Whether the types are `js.Any` or not has little to do with the decisions one should make.

Basically, the carefulness about exports is not so much about what *types* appear in the signatures you export. It's what method/class one exports that matters. Most general-purpose libraries should not export anything. But one should export a program's entry points to JavaScript code when that is required. Yes, that will cause everything reachable from there to be included in the binaries, but the types in the signatures are not the main issue.

**Comment by OlivierBlanvillain** [comment]
I misinterpreted what was going on, indeed, the type of what is exported does not matter (you can forget what I wrote on the PR, it does not make sense).

I got surprised when I included a new library to my build, and it resulted in big increase in the size of my binary, even before using it at all. Turns out the library had a few @JSExport around which caused most of the library (and parts of the stdlib) to be reachable independently of my usage.

Maybe something like the following could help clear out some misconceptions (the page on dependencies could be a better fit):?

> Adding Scala.js dependencies does not increase the size of the final binary by itself. Instead, thanks to dead code elimination, only the parts of the library which are actually used end up being packaged in the final binary. (Note that annotating parts of a library with @JSExport could negate this effect and force portions of code to be included independently of their usage.)

**Comment by sjrd** [comment]
Closing of old age ...
